proof/bulk_addition.saw

```diff
@@ -18,44 +18,115 @@ include "proof-helpers.saw";
 include "cryptol_imports.saw";
 include "vect.saw";
 include "fp_overrides.saw";
-include "fp2_overrides.saw";
 include "curve_operations.saw";
-include "curve_operations_e2.saw";
+include "ec_mult.saw";
 include "exp.saw";
+include "fp2_overrides.saw";
+include "curve_operations_e2.saw";
+include "ec_mult2.saw";
 include "exp2_for_bulk_add.saw";
+include "print_readably.saw";
+
+let do_prove=true;
+
+import "../spec/BulkAddition.cry";
+
+let {{
+bulk_addition_POINTonE1_head : POINTonE1_affine_t -> POINTonE1_affine_t -> Fp_rep_t -> [2]POINTonE1_t
+bulk_addition_POINTonE1_head a b acc = [POINTonE1_rep a', POINTonE1_rep b']
+  where
+    (a', b') = bulk_addition_head_imp Fp (POINTonE1_affine_abs a) (POINTonE1_affine_abs b) (fp_abs acc)
+bulk_addition_POINTonE2_head : POINTonE2_affine_t -> POINTonE2_affine_t -> Fp_2_rep_t -> [2]POINTonE2_t
+bulk_addition_POINTonE2_head a b acc = [POINTonE2_rep a', POINTonE2_rep b']
+  where
+    (a', b') = bulk_addition_head_imp Fp_2 (POINTonE2_affine_abs a) (POINTonE2_affine_abs b) (fp2_abs acc)
+
+bulk_addition_head_fp = bulk_addition_head_imp Fp
+bulk_addition_tail_fp = bulk_addition_tail Fp
+bulk_addition_tail_bz_fp = bulk_addition_tail_bz Fp
+stuff_fp =
+  { field = Fp
+  , bulk_addition_head_fn = bulk_addition_head_fp
+  , bulk_addition_tail_fn = bulk_addition_tail_fp
+  , bulk_addition_tail_bz_fn = bulk_addition_tail_bz_fp
+  , point_dadd_affine_fn = point_dadd_affine_fp
+  }
+
+bulk_addition_head_fp2 = bulk_addition_head_imp Fp_2
+bulk_addition_tail_fp2 = bulk_addition_tail Fp_2
+bulk_addition_tail_bz_fp2 = bulk_addition_tail_bz Fp_2
+stuff_fp2 =
+  { field = Fp_2
+  , bulk_addition_head_fn = bulk_addition_head_fp2
+  , bulk_addition_tail_fn = bulk_addition_tail_fp2
+  , bulk_addition_tail_bz_fn = bulk_addition_tail_bz_fp2
+  , point_dadd_affine_fn = point_dadd_affine_fp2
+  }
+}};
+
+let do_prove = false;
```

Comment:
Is `do_prove` supposed to be `false` here?

Reply:
yes, for now :-)